client/my-sites/post-selector/selector.jsx

```diff
@@ -85,24 +89,31 @@ const PostSelectorPosts = React.createClass( {
 		}, SEARCH_DEBOUNCE_TIME_MS );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( this.props.query, nextProps.query ) ) {
+			this.setState( {
+				requestedPages: [ 1 ]
```

Comment:
>Oh, the reason I ask is it is being set in `getInitialState`

Oh, right, forgot about that. Hmm, this seems a little fragile / redundant, wonder if we can clean it up so we're relying exclusively on the incoming query prop. I imagine it's there to keep the UI "live" with the search entry, while only triggering the search request after a debounce.

>What about adding a check for equality of `this.props.siteId !== nextProps.siteId`

For sure, makes sense.